File: $12gmall-realtime/gmall_realtime/src/main/scala/com/demo/app/BaseApp.scala
package com.demo.app

import org.apache.spark.streaming.StreamingContext

import scala.util.control.{BreakControl, Breaks}

/**
 * @author sijue
 * @date 2021/5/31 18:35
 * @describe
 */
abstract class BaseApp {
  //声名三个抽象属性
  var appName: String;
  var internal: Int;
  var ssc: StreamingContext = null;

  //控制抽象，传入处理数据的过程函数
  def process(op: => Unit) {
    try {
      op
      ssc.start();
      ssc.awaitTermination();
    } catch {
      case ex: Exception => println(ex.getMessage)
    } finally {
      ssc.stop(true, true);
    }
  }

}

File: $12gmall-realtime/gmall_realtime/readme/demo/app/BaseApp.java
//package com.demo.app;
//
//import lombok.extern.slf4j.Slf4j;
//import org.apache.spark.SparkConf;
//import org.apache.spark.streaming.Durations;
//import org.apache.spark.streaming.api.java.JavaStreamingContext;
//
///**
// * @author sijue
// * @date 2021/5/28 9:08
// * @describe
// */
//@Slf4j
//public abstract  class BaseApp {
//
//
//    /**
//     * 创建sparkContext
//     * @param appName 创建每个程序的appName不一样
//     * @param interval 每个程序的每个批次之间的间隔也可能不一样
//     * @return
//     */
//    public  JavaStreamingContext createSparkStreaming(String appName,int interval){
//          SparkConf  conf = new SparkConf().setMaster("local[4]").setAppName(appName);
//        JavaStreamingContext javaStreamingContext=new JavaStreamingContext(conf, Durations.seconds(interval));
//        return javaStreamingContext;
//    }
//    public  void  stopTertina(JavaStreamingContext javaStreamingContext)  {
//        try {
//            javaStreamingContext.start();
//            javaStreamingContext.awaitTermination();
//        }catch (Exception e){
//            log.info("spark start error");
//        }finally {
//            javaStreamingContext.close();
//        }
//
//    }
//    public abstract void process(String appName,int interval);
//}
